### DISCUSSION TITLE
How can I filter by a value inside a pivot table?

### DISCUSSION THREAD

**Discussion by lumenwrites** [supabase/supabase#6718] [Questions]
Hi! I'm building a bike rental app where users can reserve bikes for a certain number of days. Bikes and users have a many-to-many relationship, there's a bike_user pivot table that contains the information about reservation start and end dates. See the diagram:
![bikerr](https://user-images.githubusercontent.com/4761235/167254623-e74cc0ce-7cd7-4487-a0a0-a3ef499b6c41.png)

I need to take a range of dates, and filter the bikes, showing only the ones that haven't been reserved between these two dates.

Can you please help me figure out how to do this?

**Comment by laurenceisla** [comment]
Hi! If I understand correctly, you need to filter only the bikes that do not have the `reservation_start_date` in a range of two dates given by the user, right? e.g: to find the bikes with reservation start date outside of the range `2022-05-01 11:00:00` - `2022-05-02 15:00:00`, the query would be:

```javascript
const { data, error } = await supabase
  .from('bikes')
  .select('*, bike_user!inner(*)')
  .or('reservation_start_date.lt.2022-05-01 11:00:00,reservation_start_date.gt.2022-05-02 15:00:00', { foreignTable: "bike_user" })
```
